js/private/js_image_layer.bzl

```diff
@@ -248,17 +252,18 @@ def _js_image_layer_impl(ctx):
     if len(ownersplit) != 2 or not ownersplit[0].isdigit() or not ownersplit[1].isdigit():
         fail("owner attribute should be in `0:0` `int:int` format.")
 
-    default_info = ctx.attr.binary[0][DefaultInfo]
-    runfiles_dir = _runfiles_dir(ctx.attr.root, default_info)
+    binary_default_info = ctx.attr.binary[0][DefaultInfo]
+    binary_label = ctx.attr.binary[0].label
 
-    executable = default_info.files_to_run.executable
-    executable_path = paths.replace_extension(paths.join(ctx.attr.root, executable.short_path), "")
-    real_executable_path = _runfile_path(ctx, executable, runfiles_dir)
-    launcher = _write_laucher(ctx, real_executable_path)
+    binary_path = paths.join(ctx.attr.root, binary_label.package, binary_label.name)
```

Comment:
nit: this might not be safe if the the js_image_layer is in the root BUILD.bazel

Reply:
I believe paths.join will ignore empty strings but I'll double check